aws-api/src/main/java/com/amplifyframework/api/aws/GsonGraphQLResponseFactory.java

```diff
@@ -47,6 +51,10 @@
                 new GsonBuilder()
                 .registerTypeAdapter(List.class, new GsonListDeserializer())
                 .registerTypeAdapter(GraphQLResponse.Error.class, new GsonErrorDeserializer())
+                .registerTypeAdapter(Date.class, new GsonUtil.DateAdapter())
```

Comment:
No, this is not for compat.  This PR will actually break compatibility until the CLI gets updated.   Here's a table describing the old and new types:

| AppSync Type | JSON Type |  Old Java Type | New Java Type |
| ----------- | ----------- | ----------- | ----------- |
| AWSDate      | String  | java.util.Date | com.amplifyframework.core.model.AWSDate |
| AWSDateTime   | String | java.util.Date | com.amplifyframework.core.model.AWSDateTime |
| AWSTime | String | java.sql.Time | com.amplifyframework.core.model.AWSTime | 
| AWSTimestamp | Number | Long | java.util.Date |

Specifically, for current customers using AWSDate / AWSDateTime, this PR will break serialization because it will try to serialize a String to a java.util.Date, but the registered adapter for java.util.Date now expects a Number. (the previous one expected a String).

The implementation serializes AWSTimestamp directly as a java.util.Date.  I decided to go with this instead of wrapping `java.util.Date` with a custom `AWSTimestamp` object, mainly because java.util.Date already represents an AWSTimestamp (both are just time since epoch).

Reply:
This is a super useful note and table. Thank you!